src/main/java/org/opentripplanner/api/mapping/LegMapper.java

```diff
@@ -100,12 +110,76 @@ else if (domain.pathway) {
         );
         api.boardRule = domain.boardRule;
         api.alightRule = domain.alightRule;
+
+        api.pickupBookingInfo = toApi(domain.pickupBookingInfo, true);
+        api.dropOffBookingInfo = toApi(domain.dropOffBookingInfo, false);
+
         api.rentedBike = domain.rentedVehicle;
         api.walkingBike = domain.walkingBike;
 
         return api;
     }
 
+    private static ApiBookingInfo toApi(BookingInfo info, boolean isPickup) {
```

Comment:
I did not catch this the first time I looked at this, but you should make a mapper pr domain class - the mapping of BookingInfo should not be in the LegMapper - this would make it reusable as well.